src/commands/mail/address/update.tsx

```diff
@@ -0,0 +1,249 @@
+      command:
+        "<%= config.bin %> <%= command.id %> --address foo@bar.example --forward-to bar@bar.example --forward-to baz@bar.example",
+    },
+  ];
+
+  protected async setAddress(
+    mailAddressId: string,
+    address: string,
+    process: ProcessRenderer,
+  ): Promise<void> {
+    await process.runStep("set the address of a mail address", async () => {
+      const response = await this.apiClient.mail.updateMailAddressAddress({
+        mailAddressId: mailAddressId,
+        data: {
+          address: address,
+        },
+      });
+
+      assertStatus(response, 204);
+    });
+  }
+
+  protected async setCatchAll(
+    mailAddressId: string,
+    catchAll: boolean,
+    process: ProcessRenderer,
+  ): Promise<void> {
+    await process.runStep("set the catchall of a mail address", async () => {
+      const response = await this.apiClient.mail.updateMailAddressCatchAll({
+        mailAddressId: mailAddressId,
+        data: {
+          active: catchAll,
+        },
+      });
+
+      assertStatus(response, 204);
+    });
+  }
+
+  protected async setForwardAddresses(
+    mailAddressId: string,
+    forwardAddresses: string[],
+    process: ProcessRenderer,
+  ): Promise<void> {
+    await process.runStep(
+      "set the forward addresses of a mail address",
+      async () => {
+        const response =
+          await this.apiClient.mail.updateMailAddressForwardAddresses({
+            mailAddressId: mailAddressId,
+            data: {
+              forwardAddresses: forwardAddresses,
+            },
+          });
+
+        assertStatus(response, 204);
+      },
+    );
+
+    return;
+  }
+
+  protected async setPassword(
+    mailAddressId: string,
+    password: string,
+    process: ProcessRenderer,
+  ): Promise<void> {
+    await process.runStep("set the password of a mail address", async () => {
+      const response = await this.apiClient.mail.updateMailAddressPassword({
+        mailAddressId: mailAddressId,
+        data: {
+          password: password,
+        },
+      });
+
+      assertStatus(response, 204);
+    });
+  }
+
+  protected async setQuota(
+    mailAddressId: string,
+    quotaInMebiBytes: number,
+    process: ProcessRenderer,
+  ): Promise<void> {
+    await process.runStep("set the quota of a mail address", async () => {
+      const response = await this.apiClient.mail.updateMailAddressQuota({
+        mailAddressId: mailAddressId,
+        data: {
+          quotaInBytes: quotaInMebiBytes * 1024 * 1024,
+        },
+      });
+
+      assertStatus(response, 204);
+    });
+  }
+
+  protected async exec(): Promise<UpdateResult> {
+    const process = makeProcessRenderer(this.flags, "Updating a mail address");
+    const mailAddressId = await withMailAddressId(
+      this.apiClient,
+      Update,
+      this.flags,
+      this.args,
+      this.config,
+    );
+
+    if (this.flags.address) {
+      this.setAddress(mailAddressId, this.flags.address, process);
```

Comment:
Oh, I just noticed this right now! 😱   All these `set[...]` functions are `async`, but none of them are `awaited` anywhere.

I'm wondering why the linter didn't catch this... 🤔

Reply:
I saw a few cases of "missing await" warnings (of Phpstorm) in multiple files, but ignored them, because I am not very knowledgable at that topic. I fixed it know in all the mail and domain files.